functions/src/scripts/converter.ts

```diff
@@ -0,0 +1,336 @@
+    //get current cve history tree
+    const db = admin.database();
+    const ref = db.ref('/CVE_History');
+    let result = null;
+    ref.once('value', function (snapshot) {
+        result = snapshot.toJSON();
+        buildCVEHistoryTree(result, version_number, tree);
+    }).catch(error => { console.log(error) });
+}
+
+
+function buildCVEHistoryTree(tree: any, versionHistory: string, CVETree: any): void {
+    //add new version sub section to cve history if needed 
+    const longVersionNumberHistory = "BulletinVersion:" + versionHistory;
+    const db = admin.database();
+    const ref = db.ref('/CVE_History');
+    if (tree === null) {
+        //if no data present in history tree, send whole reconfigured cve tree
+        for (const ID in CVETree) {
+            const tempVersion = CVETree[ID]['BulletinVersion'];
+            delete CVETree[ID]['BulletinVersion'];
+            const tempLongVersionNumber = "BulletinVersion:" + tempVersion;
+            ref.child(ID).child(tempLongVersionNumber).set(CVETree[ID]).catch(error => { console.log(error) });
+        }
+    } else {
+        for (const CVE in CVETree) {
+            if (!Object.values(CVETree).includes(CVE)) {
+                const tempVersion = CVETree[CVE]['BulletinVersion'];
+                delete CVETree[CVE]['BulletinVersion'];
+                const tempLongVersionNumber = "BulletinVersion:" + tempVersion;
+                ref.child(CVE).child(tempLongVersionNumber).set(CVETree[CVE]).catch(error => { console.log(error) });
+            }
+        }
+        for (const CVE in tree) {
+            //check if version in history tree
+            const currentTreeCVEData = tree[CVE];
+            let hasKey = false;
+            if (JSONData[CVE] !== undefined) {
+                for (const key of Object.keys(currentTreeCVEData)) {
+                    if (key === longVersionNumberHistory) {
+                        hasKey = true;
+                    }
+                }
+                if (!hasKey) {
+                    //if doesn't have version add in the data
+                    const cveData: Record<string, object> = {};
+                    cveData['published_date'] = JSONData.published;
+                    for (const key of Object.keys(JSONData[CVE])) {
+                        cveData[key] = JSONData[CVE][key];
+                    }
+                    const constructedData = JSON.parse(JSON.stringify(cveData));
+                    delete constructedData['BulletinVersion'];
+                    ref.child(CVE).child(longVersionNumberHistory).update(constructedData).catch(error => { console.log(error) });
+                }
+            }
+        }
+    }
+    console.log("CVE History tree built");
+}
+
+function writeToDatabaseCVETree(data: any): void {
+    const db = admin.database();
+    const newRef = db.ref('/CVEs');
+    for (const ID in data) {
+        newRef.child(ID).set(data[ID]).catch(error => { console.log(error) });
```

Comment:
try to provide some additional context around the errors before logging them to console, to help make it easier for the script runner to debug